src/main/java/com/pnudev/librarysystem/mapper/UserMapper.java

```diff
@@ -0,0 +1,13 @@
+package com.pnudev.librarysystem.mapper;
+
+import com.pnudev.librarysystem.security.UserDetailsImpl;
+import com.pnudev.librarysystem.entity.User;
+import org.mapstruct.Mapper;
+
+
+@Mapper(componentModel = "spring")
+public interface UserMapper {
+
+    UserDetailsImpl toUserDetails(User user);
```

Comment:
pass interface, not implementation

Reply:
Are you sure about this? So I need to implement a factory method?